tests/app/main/views/test_communications.py

```diff
@@ -11,19 +11,56 @@
 class TestCommunicationsView(LoggedInApplicationTest):
 
     def test_get_document_upload_page_for_framework(self, data_api_client):
-        data_api_client.get_frameworks.return_value = {"frameworks": []}
+        allowed_roles = ["admin", "admin-ccs-category"]
+        for role in allowed_roles:
+            data_api_client.get_frameworks.return_value = {"frameworks": []}
+            self.user_role = role
+            framework_slug = self.load_example_listing('framework_response')['frameworks']['slug']
+
+            response = self.client.get("/admin/communications/{}".format(framework_slug))
+            assert response.status_code == 200
 
+    def test_get_document_upload_page_not_allowed_for_ccs_sourcing(self, data_api_client):
+        data_api_client.get_frameworks.return_value = {"frameworks": []}
+        self.user_role = "admin-ccs-sourcing"
         framework_slug = self.load_example_listing('framework_response')['frameworks']['slug']
 
         response = self.client.get("/admin/communications/{}".format(framework_slug))
-        assert response.status_code == 200
+        assert response.status_code == 403
 
     def test_post_documents_for_framework(self, data_api_client):
         # check we are mocking s3 as we must not actually post there!
         assert isinstance(s3.S3, mock.Mock)
 
-        data_api_client.get_frameworks.return_value = {"frameworks": []}
+        allowed_roles = ["admin", "admin-ccs-category"]
+        for role in allowed_roles:
+            data_api_client.get_frameworks.return_value = {"frameworks": []}
+            self.user_role = role
+            framework_slug = self.load_example_listing('framework_response')['frameworks']['slug']
+            dummy_file = BytesIO(u'Lorem ipsum dolor sit amet'.encode('utf8'))
+
+            response = self.client.post(
+                "/admin/communications/{}".format(framework_slug),
+                data={
+                    'communication': (dummy_file, 'test-comm.pdf'),
+                    'clarification': (dummy_file, 'test-clar.pdf'),
+                }
+            )
+
+            # check that we did actually mock-send two files
+            self.s3.return_value.save.call_count == 2
+            flash_messages = self._get_flash_messages()
+            # arguably a misunderstanding here about what an appropriate 'message category' looks like
+            assert set(flash_messages.getlist('upload_communication')) == set(('clarification', 'communication',))
+
+            assert response.status_code == 302
+
+    def test_ccs_sourcing_can_not_post_documents_for_framework(self, data_api_client):
+        # check we are mocking s3 as we must not actually post there!
+        assert isinstance(s3.S3, mock.Mock)
```

Comment:
Out of interest, do we use the `moto` library at all for mocking `boto`/AWS actions? I've found it quite useful in the past as you can just add a `@mock_s3` decorator to your test class, and never worry about accidentally putting something in a real bucket ever again 😸

Reply:
I know Rob looked at it at some point but can't remember why he decided not to use it.  This is a straight copy-and-paste job from `test_post_documents_for_framework` above TBH so I've given it zero thought here.